testng-core/src/test/kotlin/test/SimpleBaseTest.kt

```diff
@@ -0,0 +1,407 @@
+        fun run(vararg suites: XmlSuite): InvokedMethodNameListener = run(false, *suites)
+
+        @JvmStatic
+        fun run(skipConfiguration: Boolean, vararg suites: XmlSuite) =
+            run(skipConfiguration, create(*suites))
+
+        @JvmStatic
+        protected fun create(outputDir: Path, vararg testClasses: Class<*>) =
+            create(*testClasses).apply {
+                outputDirectory = outputDir.toAbsolutePath().toString()
+            }
+
+        @JvmStatic
+        protected fun create(vararg suites: XmlSuite) = create(listOf(*suites))
+
+        @JvmStatic
+        protected fun create(suites: List<XmlSuite>) = create().apply {
+            setXmlSuites(suites)
+        }
+
+        @JvmStatic
+        protected fun create(outputDir: Path, vararg suites: XmlSuite) =
+            create(outputDir, listOf(*suites))
+
+        @JvmStatic
+        protected fun create(outputDir: Path, suites: List<XmlSuite>) = create(suites).apply {
+            outputDirectory = outputDir.toAbsolutePath().toString()
+        }
+
+        @JvmStatic
+        protected fun createTests(suiteName: String, vararg testClasses: Class<*>) =
+            createTests(null, suiteName, *testClasses)
+
+        @JvmStatic
+        protected fun createTests(
+            outDir: Path?,
+            suiteName: String,
+            vararg testClasses: Class<*>
+        ) = createXmlSuite(suiteName).let { suite ->
+            for ((i, testClass) in testClasses.withIndex()) {
+                createXmlTest(suite, testClass.name + i, testClass)
+            }
+            //if outDir is not null then create suite with outDir, else create suite without it.
+            outDir?.let { create(it, suite) } ?: create(suite)
+        }
+
+        @JvmStatic
+        protected fun createDummySuiteWithTestNamesAs(vararg tests: String) =
+            XmlSuite().apply {
+                name = "random_suite"
+                tests.forEach {
+                    XmlTest(this).apply {
+                        name = it
+                    }
+                }
+            }
+
+        @JvmStatic
+        protected fun createXmlSuite(name: String) = XmlSuite().apply {
+            this.name = name
+        }
+
+        @JvmStatic
+        protected fun createXmlSuite(params: Map<String, String>) =
+            createXmlSuite(UUID.randomUUID().toString()).apply {
+                parameters = params
+            }
+
+        @JvmStatic
+        protected fun createXmlSuite(
+            suiteName: String,
+            testName: String,
+            vararg classes: Class<*>
+        ) = createXmlSuite(suiteName).apply {
+            createXmlTest(this, testName, *classes)
+        }
+
+        @JvmStatic
+        protected fun createXmlSuite(suiteName: String, params: Map<String, String>) =
+            createXmlSuite(suiteName).apply {
+                parameters = params
+            }
+
+        @JvmStatic
+        protected fun createXmlTestWithPackages(
```

Comment:
From a Kotlin test, I think `xmlSuite.createXmlTestWithPackages(...)` is more logical than `createXmlTestWithPackages(xmlSuite, ...`.

You can fix it by adding an extension.

Same logic for the other facture functions.

Reply:
@juherr - From what I have read about extension functions in Kotlin, they get translated into `static` funtions in the Java world, with the type (to which they are added as extension funtions) getting passed implicitly as the first parameter to the static method.

The problem with this is that now all the Java sub classes will be required to start adding the extra static parameter which is going to add a whole lot of changes.

So can we for now just stick to the `@JvmStatic` way of exposing the companion methods and then move forward with migrating the tests into Kotlin. I will take care of incremental refactoring of the tests to be more Kotlinish, once the migration is complete, because that is going to be a lot more easier.